Automattic/wp-calypso
# Signup: create site preview iframe component

## Changes proposed in this Pull Request
This PR introduces an asynchronously-loaded component that encapsulates a site preview, with mocked content and current theme CSS for both mobile and phone views.

![Apr-10-2019 17-13-44](https://user-images.githubusercontent.com/6458278/55858844-0c4b1980-5bb4-11e9-83d1-196e89719288.gif)

This works by creating an iframe containing a WordPress page HTML shell into which we pump:

1. a theme CSS URI
2. a Google Font CSS URI
3. title, tagline and body content

## TODO
- [x] Make it work in IE11 :trollface: 
<img width="500" alt="Screen Shot 2019-04-09 at 5 38 06 pm" src="https://user-images.githubusercontent.com/6458278/55781794-4d78f600-5aee-11e9-95c3-b7981f73dfcd.png">

## Testing instructions

Fire up the branch, test the preview. In particular, play around with http://calypso.localhost:3000/start/onboarding-dev/site-style-with-preview

Preview in different widths, throttle the network, try to break it.

For bonus points, apply the yet-unfinished vertical annotation localization patch `D26776-code` and switch to a `RTL` language such as Hebrew or Arabic.

The iframe HTML tag should update to something like: `<html lang="he" dir="rtl">` and look right-to-left:

<img width="1258" alt="Screen Shot 2019-04-12 at 12 36 56 pm" src="https://user-images.githubusercontent.com/6458278/56008578-fce8df00-5d1f-11e9-8ce7-66b54b36571b.png">
